PrincipalController.java

```diff
@@ -1,56 +1,51 @@
 package com.example.demo;
+
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.HttpStatus;
 import java.util.*;
 import javax.servlet.http.*;
-import org.springframework.http.server.reactive.ServerHttpRequest;
-import javax.servlet.Filter;
-import javax.servlet.FilterChain;
-import javax.servlet.FilterConfig;
-import javax.servlet.ServletException;
-import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
-
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.*;
 import java.io.*;
 import org.springframework.stereotype.Component;
 @RestController
 
 public class PrincipalController {
-	ParkingLot pl1 = new ParkingLot();
+	@Autowired
+	ParkingLot pl1;

```

Comment:
let's name our variable a little better - it makes the code more readable - change `pl1` to `parkingLot`